newScript.js

```diff
@@ -0,0 +1,295 @@
+const main = document.getElementsByTagName("main")[0];
+const form = document.getElementById("form");
+const subBtn = document.getElementById("subButton");
+const deleteAll = document.getElementById("deleteBtn");
+const seeMoreBtn = document.getElementById("seeMoreBtn");
+const seeMoreDiv = document.getElementById("seeMoreDiv");
+const input = document.getElementById("input");
+const dateInput = document.getElementById("date");
+const tagInput = document.getElementById("tag");
+const addFirstBtn = document.getElementById("addFirst");
+const addRandom = document.getElementById("addRandomPos");
+const removeFirst = document.getElementById("removeFirst");
+const removeLast = document.getElementById("removeLast");
+const removeRandom = document.getElementById("removeRandom");
+const removeDup = document.getElementById("removeDuplicates");
+const removeOdd = document.getElementById("removeOdd");
+const randomize = document.getElementById("randomize");
+const alphaBtn = document.getElementById("alphabetically");
+const alphaRvrsBtn = document.getElementById("alphabeticallyRvrs");
+
+
+
+let list = document.getElementById("list");
+let listToDo = JSON.parse(localStorage.getItem("list"));
+
+
+
+if (listToDo?.length === 0 || !listToDo) {
+  listToDo = [];
+}
+
+function updatePage() {
+  list.innerHTML = null;
+  for (let i = 0; i < listToDo.length; i++) {
+    createOnPg(listToDo[i].taskName, listToDo[i].date, listToDo[i].tag);
+  }
+  refreshLocalStorage(listToDo);
+}
+
+form.addEventListener("submit", (e) => {
+  e.preventDefault();
+});
+
+input.addEventListener("focus", () => {
+  dateInput.setAttribute("id", "date-appear");
+  setTimeout(() => { tagInput.setAttribute("id", "tag-appear") }, 300);
+});
+
+removeOdd.addEventListener("click", () => {
+  for (let i = 0; i < listToDo.length; i++) {
+    if (i % 2 === 0) {
+      listToDo.splice(i, 1);
+    }
+  }
+  updatePage();
+})
+
+removeDup.addEventListener("click", () => {
+  const tempSet = new Set();
+  const tempArray = [];
+  for (let i = 0; i < listToDo.length; i++) {
+    if (!tempSet.has(listToDo[i].taskName)) {
+      tempArray.push(listToDo[i]);
+    }
+    tempSet.add(listToDo[i].taskName);
+  }
+  listToDo = tempArray;
+  updatePage();
+});
+
+removeRandom.addEventListener("click", () => {
+  const random = Math.round(Math.random() * listToDo.length - 1);
+  listToDo.splice(random, 1);
+  updatePage();
+});
+
+removeLast.addEventListener("click", () => {
+  listToDo.pop();
+  updatePage();
+});
+
+removeFirst.addEventListener("click", () => {
+  listToDo.splice(0, 1);
```

Comment:
Está ok, mas o JavaScript é um fofo e já tem um método para isso :)
```suggestion
  listToDo.shift();
```
https://www.w3schools.com/jsref/jsref_shift.asp